File: java_solutions/43.multiply-strings.java
/*
 * @lc app=leetcode id=43 lang=java
 *
 * [43] Multiply Strings
 */

// @lc code=start
// semicolons : 1
import java.math.*;

class Solution {
    public String multiply(String num1, String num2) {
        return Stream.of(num1, num2)
                     .map(BigInteger::new)
                     .reduce(BigInteger.ONE, BigInteger::multiply)
                     .toString();
    }
}
// @lc code=end

File: kotlin_solutions/43.multiply-strings.kt
/*
 * @lc app=leetcode id=43 lang=kotlin
 *
 * [43] Multiply Strings
 */

class Solution {fun multiply(num1: String, num2: String): String {return (java.math.BigInteger(num1)*java.math.BigInteger(num2)).toString()}}

